tests/cpgrid/addLgrsOnDistributedGrid_test.cpp

```diff
@@ -753,6 +876,57 @@ BOOST_AUTO_TEST_CASE(distributed_in_all_ranks_lgr)
         // 64 new refined cells. 5x5x5 = 125 points (only 98 = 125 - 3x3x3 parent corners new points - new global ids).
         grid.addLgrsUpdateLeafView(cells_per_dim_vec, startIJK_vec, endIJK_vec, lgr_name_vec);
         refinePatch_and_check(grid, cells_per_dim_vec, startIJK_vec, endIJK_vec, lgr_name_vec);
+
+        // Check global id is not duplicated for points for each LGR
+        // LGR1 dim 4x4x4 -> 5x5x5 = 125 points
+        std::vector<int> local_point_ids;
+        local_point_ids.reserve(125); // expected_point_ids in LGR1
+        for (const auto& element : elements(grid.levelGridView(1))) {
+            for (int corner = 0; corner < 8; ++corner)
+            {
+                const auto& point = element.subEntity<3>(corner);
+                local_point_ids.push_back(grid.currentData()[1]->globalIdSet().id(point));
+            }
+        }
+        auto [all_point_ids, displPoint ] = Opm::allGatherv(local_point_ids, grid.comm());
+        const std::set<int> all_point_ids_set(all_point_ids.begin(), all_point_ids.end());
+        // Difference with preveious test case:
+        // Global cell ids of cells to be refined = {1,2,5,6,13,14,17,18}
```

Comment:
Please indicate that this a bug originating from the way that we add overlap cells.